apps/jobs/views.py

```diff
@@ -933,6 +934,71 @@ def update_submission(request, challenge_pk):
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
+@api_view(['POST', ])
+@throttle_classes([UserRateThrottle, ])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def re_run_submission(request, submission_id):
+    """
+    API endpoint to re-run a submission.
+    Only challenge host has access to this endpoint.
+    """
+    try:
+        submission = Submission.objects.get(id=submission_id)
+        logger.info('Submission found with submission number {}'
+                    .format(submission_id))
+    except Submission.DoesNotExist:
+        response_data = {'error': 'Submission with submission number {} does not exist'.format(submission_id)}
+        return Response(response_data, status=status.HTTP_404_NOT_FOUND)
+
+    # check if the challenge phase exists or not
+    try:
+        challenge_phase = submission.challenge_phase
+        challenge_phase_id = challenge_phase.id
+    except Exception as e:
+        response_data = {'error': 'Challenge Phase does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # check if the challenge exists or not
+    try:
+        challenge = challenge_phase.challenge
+        challenge_id = challenge.id
+    except Exception as e:
+        response_data = {'error': 'Challenge does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if not is_user_a_host_of_challenge(request.user, challenge_id):
+        response_data = {
+            "error": "You are not allowed to re-run the challenge submission"
+        }
+        return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    if not challenge.is_active:
+        response_data = {'error': 'Challenge is not active'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # check if challenge phase is active
+    if not challenge_phase.is_active:
+        response_data = {
+            'error': 'Sorry, cannot accept submissions since challenge phase is not active'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+    try:
+        publish_submission_message(challenge_id, challenge_phase_id, submission.id)
+        time.sleep(1)
```

Comment:
Why sleep?

Reply:
@galipremsagar actually when `publish_submission_message` is called it needs some time to complete the submission and if we don't sleep it for 1 sec then we won't get the updated submission object to show it dynamically on the page instead we'll get the submission object with the previous submission status.

Reply:
At other places there seems to be no sleep after publish_submission_message, can you test it out and confirm if we really need sleep? I'm okay with keeping it if necessary though but just need to confirm if we really need it.

Reply:
@galipremsagar I think at other places like at the time making submission we don't need the submission object or it's fields to render dyanmically

Reply:
I don't think we need this sleep thing as the submission status is `Submitted` when it is pushed to a queue.